ssz_serialization/merkleization.nim

```diff
@@ -65,17 +65,26 @@ type
     # `sha256.update` otherwise would do.
     # The two digests represent the left and right nodes that get combined to
     # a parent node in the tree.
+    # `SszMerkleizer` used chunk count as limit
     # TODO it's possible to further parallelize by using even wider buffers here
 
-    combinedChunks: array[binaryTreeHeight limit, (Digest, Digest)]
+    combinedChunks: array[height, (Digest, Digest)]
     totalChunks*: uint64 # Public for historical reasons
     topIndex: int
     internal: bool
+      # Avoid copying chunk data into merkleizer when not needed - maw result
+      # in an incomplete root-to-leaf proof
 
-template getChunkCount*(m: SszMerkleizer): uint64 =
+template limit*(T: type SszMerkleizer2): Limit =
+  if T.height == 0: 0'i64 else: 1'i64 shl (T.height - 1)
```

Comment:
Hrm, what about the cases where the limit is not a power of 2? We have to review more carefully how this code is being used.

Reply:
no critical usages, ie it doesn't materially matter since all trees are extended to power-of-2 anyway